Sample_Program1.py

```diff
@@ -0,0 +1,73 @@
+class Sorcery:
+    'Common base class for magic spells'
+    sorceryCount = 0
+
+    def __init__(self, name, MPCost, element):
+        self.name = name
+        self.MPCost = MPCost
+        self.element = element
+        Sorcery.sorceryCount += 1
+
+    def displayCount(self):
+        print("Total Sorceries: %d" % Sorcery.sorceryCount)
+
+    def displayMagicAttributes(self):
+        print("Name: " + self.name + '\n' + "MP Cost: %d" % self.MPCost)
+        print("Element: " + self.element)
+
+    def getName(self):
+        return self.name
+
+# Creates some basic spells and returns a
+# list of them
+def createSpells(hippoFlag):
+    spellBook = []
+    fireBall = Sorcery("Fireball", 5, "Fire")
+    iceStorm = Sorcery("Ice Storm", 20, "Frost")
+    shadowDagger = Sorcery("Shadow Dagger", 7, "Dark")
+
+    spellBook.append(fireBall)
+    spellBook.append(iceStorm)
+    spellBook.append(shadowDagger)
+
+    if (hippoFlag):
+        summonHippo = Sorcery("Summon Hippopotamus", 50, "Arcane")
+        spellBook.append(summonHippo)
+
+    return spellBook
+
+#The printSpellBook function takes in a boolean and
+#will print a list of spells depending on what the boolean
+#is initialized to. It will not print the list if the
+#size of the spell list is 0, or if forbidden spells
+#are allowed.
+def printSpellBook(includeForbiddenSpells):
+    hippoSpellName = "Summon Hippopotamus"
+    hippoFlag = False
+    forbiddenSpellLocation = -1
+    spellBook = createSpells(includeForbiddenSpells)
+
+    if (len(spellBook) <= 0):
+        print("The spell book is currently empty;")
+        print("Please add some spells if you are wanting to print them.")
+        return
+
+    print("Checking to see if all spells in the spell book are valid...")
+    for i in range(len(spellBook)):
+        if (spellBook[i].getName() == hippoSpellName):
+            hippoFlag = True
+            forbiddenSpellLocation = i
+
+    spellBook[0].displayCount()
+    if (hippoFlag):
+        print("The hippopotamus spell is too powerful. It")
+        print("should not be added to the spellBook!")
+        print("Here's some info on it anyway:")
```

Comment:
Rather than renaming the Sample_Program1.py file to Main.py, I renamed it to Sorcery_Functions.py. I'm not sure if that's what you suggested, but there you have it. I then created a Main.py to use to test the Sorcery_Functions.py file. The Main.py file is basically a script meant for testing right now, but we can change the functionality of it later if you want. In the Main.py file I already imported the Sorcery_Functions module, and after the import I added some test cases so that testing it is a bit more streamlined. Therefore, to execute the tests you simply have to import the Main.py file using the Python command line, and then run the function testSorceries() in the Main module. So I guess that Main.py is really more of a module than a script since it has a function definition, and you have to import the module and run a function from it, but I guess it's whatever you want to call it.

Let me know if you have issues testing the Sorcery_Functions module.

Reply:
One more thing I forgot to mention. I know you said to add spaces to the end of my print() statements, but I'm not entirely sure what the point is because, for my interpreter anyway, the print() function automatically adds a new line after printing the text inputted into it. Was there a specific reason to add spaces at the end of the print() statements? I can definitely see it being an issue if I were chaining together output text onto a single line with multiple print() statements, but I'm not really sure if you can even do that in Python.